compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -79,8 +79,13 @@ trait SpaceLogic {
   /** Is `tp1` a subtype of `tp2`? */
   def isSubType(tp1: Type, tp2: Type): Boolean
 
-  /** Is `tp1` the same type as `tp2`? */
-  def isEqualType(tp1: Type, tp2: Type): Boolean
+  /** Whether we may assume the two Unapply the same?
+   *  That is, given the same parameter, returns the same result.
+   *
+   *  This is more general than purity, as the same `unapply` method may
+   *  take different prefix, thus behaves differently.
```

Comment:
I was a bit confused by this description, here's a proposed slightly different version:
```suggestion
  /** True if we can assume that the two unapply methods are the same.
   *  That is, given the same parameter, they return the same result.
   *
   *  We assume that unapply methods are pure, but the same method may
   *  be called with different prefixes, thus behaving differently.
   */
```